src/emit/mod.rs

```diff
@@ -5062,10 +5077,10 @@ pub trait TargetRuntime<'a> {
                         bin.builder.build_int_add(len, len, "hex_len")
                     }
                     Type::Uint(bits) if *spec == FormatArg::Hex => {
-                        bin.context.i32_type().const_int(bits as u64 / 8 + 2, false)
+                        bin.context.i32_type().const_int(bits as u64 / 4 + 2, false)
```

Comment:
Because it causes a buffer overrun. A converting an unsigned int to a hex string with the `0x` prefix can be a maximum of `bits / 4 + 2`, characters, not `bits / 8 + 2`.